### ISSUE TITLE
Don't play Join/Leave sound if Bot is already in channel/server

### ISSUE THREAD

**Issue by Cazed** [tmercswims/tmerc-cogs#80]
Hi,

I have been using your custom join/leave sounds cog for a little while now and I am having a problem with it when the bot is playing audio already in the server.

The audio goes all weird (I assume as it is trying to play two audio sources at once) and then disconnects after the join/leave sound has finished, without letting it finish the music.

I've tried to change a few parts of the code, but as a Programming newbie, I am struggling.

Any help is hugely appreciated.

Jake

**Comment by tmercswims** [comment]
You're right, the sound is distorting because it's trying to play two things at the same time. As weird as it might sound, the real solution is to install an additional cog, [SFX](https://cogs.red/cogs/flapjax/FlapJack-Cogs/sfx/) by @flapjax.

His SFX cog can interrupt playing audio to play a sound bite, and then the normal audio will resume. It also exposes a way for other cogs to take advantage of this functionality, which CJL does (if the cog is installed).

So give that a try, if you don't mind, and let me know if it fixes the issue. If you want to discuss things further, feel free to join my cog support channel [here](https://discord.gg/HWfwYxN).

**Comment by Cazed** [comment]
Hi Dude,

Sorry for the delay, real life got in the way!

Installing the SFX cog seems to have resolved the issue entirely. Thank you very much!

**Comment by tmercswims** [comment]
Great to hear! Thanks for following up 😄
